h2o-core/src/main/java/water/rapids/ast/prims/assign/AstRectangleAssign.java

```diff
@@ -162,25 +190,14 @@ private void assign_frame_scalar(Frame dst, int[] cols, final AstNumList rows, f
     for (int i = 0; i < cols.length; i++)
       vecs2[i] = vecs[cols[i]];
     rows.sort();                // Side-effect internal sort; needed for fast row lookup
-    new MRTask() {
+    new RowSliceTask(rows) {
```

Comment:
The same as above - it is probably better to have a method returning the instance of task, or have a static type.